### PR TITLE
Fix 111 - support populate for patch and update

### PR DESCRIPTION
**Pull request by ekryski**

### CODE REVIEW

File: src/service.js [hunk 1]
```diff
@@ -125,18 +127,23 @@ class Service {
...
   }
 
   update(id, data, params) {
-    if(id === null) {
+    if (id === null) {
       return Promise.reject('Not replacing multiple records. Did you mean `patch`?');
     }
 
+    // Handle case where data might be a mongoose model
+    if (typeof data.toObject === 'function') {
```
Comment (ekryski) [inline src/service.js]: I accidentally committed code from the other PR but for some reason even though I cleaned up the commits the diff is still showing the old stuff.

File: src/service.js [hunk 2]
```diff
@@ -188,8 +207,16 @@ class Service {
     // We need this shitty hack because update doesn't return
     // a promise properly when runValidators is true. WTF!
     try {
+      let query = Object.assign({}, params.query);
+
+      // If params.query.$populate was provided, remove it
+      // from the query sent to mongoose.
+      if (query.$populate) {
+        delete query.$populate;
```
Comment (daffl) [inline src/service.js]: Needs to be `query = omit(query, '$populate')` as to not modify the original object.

Comment (ekryski) [inline src/service.js]: @daffl I'm copying it above using object.assign. But that's cleaner.

**Comment by ekryski** [comment]
Manually merged due to my commit fuckery.

**Pull request closed without merging**